### PR TITLE
WIP: Add hook information to the log

### LINKED ISSUE #696: show on-run-start/end + hooks in dbt output
### Feature

dbt doesn't output any info about the start/end hooks that it runs, so it can be hard to understand why the stdout seems to stall. Instead, dbt should log the start/end hooks that it is running to the console.

Example:
```
$ dbt run

Found 3 models, 2 tests, 0 archives, 0 analyses, 96 macros, 2 operations, 0 seed files

12:27:16 | Concurrency: 8 threads (target='dev')
12:27:16 |
12:27:16 | Running 3 on-run-start hooks:
12:27:16 | 1 of 3 START grant usage on ...
12:27:16 | 2 of 3 START grant create on ...
12:27:16 | 3 of 3 START insert into public._audit ...
12:27:16 |
12:27:16 | Running 2 models:
12:27:16 | 1 of 2 START table model demo_schema.dep............................. [RUN]
12:27:16 | 2 of 2 START incremental model demo_schema.local..................... [RUN]
12:27:16 | 1 of 2 OK created table model demo_schema.dep........................ [SELECT 1 in 0.13s]
12:27:16 | 2 of 2 OK created incremental model demo_schema.local................ [INSERT 0 1 in 0.14s]
12:27:16 |
12:27:16 | Running 3 on-run-end hooks:
12:27:16 | 1 of 3 START grant usage on ...
12:27:16 | 2 of 3 START grant create on ...
12:27:16 | 3 of 3 START insert into public._audit ...
12:27:16 |
12:27:16 | Finished running 1 table models, 1 incremental models, 6 run hooks in 0.32s.

Completed successfully

Done. PASS=2 ERROR=0 SKIP=0 TOTAL=2
```

If there are no on-run-start/end hooks, dbt should skip those sections of the info logs. Note that we should also add something like `6 run hooks` to the final line of model run logs.

^ this is an initial idea of how this could look, but I don't know that I _love_ it. Very open to other ideas!

### PR DESCRIPTION
**Pull request by vijaykiran**
This commit adds logging info for on-run-start/end hooks (Fixes #696)

Here's the sample output. 

```
Found 3 models, 0 tests, 0 archives, 0 analyses, 95 macros, 7 operations, 1 seed files

21:04:16 |
21:04:16 | Running 3 on-run-start hooks:
21:04:16 | 1 of 3 select 'ON RUN START HOOK - 1'
21:04:16 | 2 of 3 select 'ON RUN START HOOK - 2'
21:04:16 | 3 of 3 select 'ON RUN START HOOK - 3'
21:04:16 | Concurrency: 4 threads (target='default')
21:04:16 |
21:04:16 | 1 of 3 START view model dbt_postgres.moby_dick_base.................. [RUN]
21:04:16 | 1 of 3 OK created view model dbt_postgres.moby_dick_base............. [CREATE VIEW in 0.05s]
21:04:16 | 2 of 3 START table model dbt_postgres.word_count..................... [RUN]
21:04:16 | 2 of 3 OK created table model dbt_postgres.word_count................ [SELECT 27390 in 0.32s]
21:04:16 | 3 of 3 START view model dbt_postgres_new.word_count_custom_schema.... [RUN]
21:04:16 | 3 of 3 OK created view model dbt_postgres_new.word_count_custom_schema [CREATE VIEW in 0.02s]
21:04:16 |
21:04:16 | Running 2 on-run-end hooks:
21:04:16 | 1 of 2 select 'ON RUN END HOOK - 1'
21:04:16 | 2 of 2 select 'ON RUN END HOOK - 2'
21:04:16 |
21:04:16 | Finished running 2 view models, 1 table models in 0.54s.
```

Adding the hook count to the final line (Finished ...) seems more invasive because as far as I understand, hooks don't return `results` object. @drewbanin any pointers on how to do this in a clean way?

### CODE REVIEW

**Comment by drewbanin** [comment]
hey @vijaykiran - thanks for this PR -- it looks like a great start. I think you're right that we will want to make hooks return a `results` object. That way, we could show the status of the query, as well as how long it took to run. 

I'm also not sure that I _love_ the output shown here (and described in the original issue). I think that for any non-trivial SQL, showing the full statement is going to be overly verbose.

Maybe we can do something like:
```
21:04:16 | Running 3 on-run-start hooks:
21:04:16 | 1 of 3 on-run-start-1................. [RUN]
21:04:16 | 1 of 3 on-run-start-1................. [SELECT in 0.05s]
21:04:16 | 2 of 3 on-run-start-2................. [RUN]
21:04:16 | 2 of 3 on-run-start-2................. [SELECT in 0.05s]
21:04:16 | 3 of 3 on-run-start-3................. [RUN]
21:04:16 | 3 of 3 on-run-start-3................. [SELECT in 0.05s]
21:04:16 |
21:04:16 | Concurrency: 4 threads (target='default')
```

File: dbt/node_runners.py [hunk 1]
```diff
@@ -348,6 +350,8 @@ def run_hooks(cls, config, adapter, manifest, hook_type, extra_context):
                 dbt.contracts.graph.parsed.Hook(**hook_dict)
 
             sql = hook_dict.get('sql', '')
+            dbt.ui.printer.print_timestamped_line("{} of {} {} ".
+                                                  format(i+1, hook_count, sql))
```
Comment (drewbanin) [inline dbt/node_runners.py]: you can construct a `RunModelResult` object here with something like:
```
res = RunModelResult(hook, status=status)
```

You can get a status from the `adapter.execute` call below. That will return a tuple of a status and an Agate dataframe, as shown here: https://github.com/fishtown-analytics/dbt/blob/dev/stephen-girard/dbt/adapters/base/impl.py#L138

eg:
```
status, _ = adapter.execute(sql, model_name=model_name, auto_begin=False, fetch=False)
```

Comment (drewbanin) [reply dbt/node_runners.py]: ^ I think this will work, but I haven't tested it. Let me know if you hit any more snags

**Comment by vijaykiran** [comment]
@drewbanin  Thanks for the feedback!

> I'm also not sure that I love the output shown here (and described in the original issue). I think that for any non-trivial SQL, showing the full statement is going to be overly verbose.

I agree and I think adding SQL become unreadable quickly. I'll update the code.  I'll dig a bit deeper into Results stuff this weekend.

**Comment by vijaykiran** [comment]
I'll open a new one with the Results work, thanks for the review @drewbanin !

**Pull request closed without merging**